### PR TITLE
[flang][openacc] Fix missing bounds for allocatable and pointer array component

### PR DESCRIPTION
**Pull request by clementval**
Bounds were not gathered correctly for pointer and allocatable array components. This patch fixes the issues pointed in https://reviews.llvm.org/D158732.
The change should also enable correct bounds gathering for the OpenMP implementation. 

A new test file `acc-bounds.f90` is added and bounds specific tests currently in `acc-enter-data.f90` can be moved there in a follow up patch.

### CODE REVIEW

**Comment by agozillon** [comment]
LGTM, please however, wait on further review from @razvanlupusoru

**Review by agozillon — APPROVED** [review]

**Review by razvanlupusoru — APPROVED** [review]
Looks great! Thank you for fixing this!

**Pull request merged**
